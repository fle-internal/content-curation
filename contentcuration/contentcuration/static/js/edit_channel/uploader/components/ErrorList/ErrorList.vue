<template>
  <VList v-if="errors && errors.length">
    <VListTile
      v-for="(error, idx) in errors"
      :key="idx"
    >
      <span class="red--text font-weight-medium">{{ error }}</span>
    </VListTile>
  </VList>
</template>

<script>

  export default {
    name: 'ErrorList',
    props: {
      errors: {
        type: Array,
      },
    },
  };

</script>

<style lang="less" scoped>

/deep/ .v-list__tile {
  padding-left: 0;
}

</style>
