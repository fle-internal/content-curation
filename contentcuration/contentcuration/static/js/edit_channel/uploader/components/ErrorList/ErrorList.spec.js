import { shallowMount, mount } from '@vue/test-utils';

import ErrorList from './ErrorList';

describe('ErrorList', () => {
  it('smoke test', () => {
    const wrapper = shallowMount(ErrorList);

    expect(wrapper.isVueInstance()).toBe(true);
  });

  it('renders nothing when there are no errors', () => {
    const wrapper = mount(ErrorList, {
      propsData: {
        errors: [],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders all errors', () => {
    const wrapper = mount(ErrorList, {
      propsData: {
        errors: ['Oops!', 'Something went wrong :('],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
