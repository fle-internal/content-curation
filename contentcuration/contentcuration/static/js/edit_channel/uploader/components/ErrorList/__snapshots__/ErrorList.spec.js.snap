// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorList renders all errors 1`] = `
<div role="list" class="v-list theme--light">
  <div role="listitem">
    <div class="v-list__tile theme--light"><span class="red--text font-weight-medium">Oops!</span></div>
  </div>
  <div role="listitem">
    <div class="v-list__tile theme--light"><span class="red--text font-weight-medium">Something went wrong :(</span></div>
  </div>
</div>
`;

exports[`ErrorList renders nothing when there are no errors 1`] = `undefined`;
